sklearn/preprocessing/tests/test_label.py

```diff
@@ -511,3 +517,44 @@ def test_inverse_binarize_multiclass():
                                                    [0, 0, 0]]),
                                        np.arange(3))
     assert_array_equal(got, np.array([1, 1, 0]))
+
+
+def test_transformed_target_regressor_error_kwargs():
+    X = friedman[0]
+    y = friedman[1]
+    # provide a transformer and functions at the same time
+    clf = TransformedTargetRegressor(estimator=LinearRegression(),
+                                     transformer=StandardScaler(),
+                                     func=np.exp, inverse_func=np.log)
+    assert_raises_regex(ValueError, "Both 'transformer' and functions"
+                        " 'func'/'inverse_func' cannot be set at the"
+                        " same time.", clf.fit, X, y)
+
+
+def test_transformed_target_regressor_invertible():
+    X = friedman[0]
+    y = friedman[1]
+    clf = TransformedTargetRegressor(estimator=LinearRegression(),
+                                     func=np.exp, inverse_func=np.exp,
+                                     check_invertible=True)
+    assert_raise_message(ValueError, "The provided functions or transformer"
+                         " are not strictly invertible.", clf.fit, X, y)
+    clf = TransformedTargetRegressor(estimator=LinearRegression(),
+                                     func=np.exp, inverse_func=np.exp,
+                                     check_invertible=False)
+    # the transformer/functions are not checked to be invertible the fitting
+    # should pass
+    clf.fit(X, y)
+
+
+def test_target_transformer_friedman():
```

Comment:
That, and as you said, you should have one with y.shape = (n_samples, 1) and one with y.shape = (n_samples, 3) or something.